inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -924,11 +930,61 @@ public KBHandle update(KnowledgeBase kb, UpdateAction aAction)
                 tupleQuery.setBinding("pDESCRIPTION", aKB.getDescriptionIri());
                 tupleQuery.setIncludeInferred(false);
     
-                return evaluateListQuery(tupleQuery, aAll, "s", "l", "d");
+                return evaluateListQuery(aKB, tupleQuery, true, aAll, "s", "l", "d");
             });
         }
-        resultList.sort(Comparator.comparing(KBObject::getUiLabel));
-        return resultList;
+        
+        List<KBHandle> resultLabelList = readLabelsWithoutLanguage(aKB, aAll, resultList);
+        resultLabelList.sort(Comparator.comparing(KBObject::getUiLabel));
+        return resultLabelList;
+    }
+    
+    private List<KBHandle> readLabelsWithoutLanguage(KnowledgeBase aKB, boolean aAll,
+            List<KBHandle> resultList)
+    {
+        List<KBHandle> resultLabelList = new ArrayList<>();        
+        for (KBHandle result : resultList) {
```

Comment:
This loop here generates **many** queries. E.g. when used on `http://zbw.eu/beta/sparql/gnd/query` it takes really long to load. Do you think it would be possible to send a single query where all IDs for which we want to have labels are included - or at least to batch-query for like 100 labels at a time?

Reply:
Yeah, I thought about it. Ideally when u have an IRI present for specific concept/instance, it's better comparatively. But yeah, there is overhead of queries in here. You wanna put in some pointers here?

Reply:
@naveen2507 What kind of pointer would you like?

Reply:
I meant, do u hv some alternative approach in mind??